services/web/server/src/simcore_service_webserver/projects/_metadata_db.py

```diff
@@ -39,18 +58,52 @@ async def get_project_metadata(engine: Engine, project_uuid: ProjectID) -> Metad
         return parse_obj_as(MetadataDict, metadata.custom or {})
 
 
+_NIL_NODE_UUID: Final[NodeID] = NodeID(int=0)
+
+
 async def set_project_metadata(
     engine: Engine,
     project_uuid: ProjectID,
     custom_metadata: MetadataDict,
+    parent_project_uuid: ProjectID | None,
+    parent_node_id: NodeID | None,
 ) -> MetadataDict:
     """
     Raises:
         ProjectNotFoundError
+        NodeNotFoundError
```

Comment:
what about DBProjectNodeParentNotFoundError, ProjectInvalidUsageError or ProjectInvalidUsageError? How are these also handled

Reply:
```DBProjectNodeParentNotFoundError``` is not raised here.
but I agree for ```ParentNodeNotFoundError``` and ```ProjectInvalidUsageError``` added